src/renderer/dx/DxRenderer.cpp

```diff
@@ -560,6 +700,8 @@ try
         CATCH_LOG(); // A failure in the notification function isn't a failure to prepare, so just log it and go on.
     }
 
+    _recreateDeviceRequested = false;

```

Comment:
Moving it here made sense to me and also avoids double-creation of the DX resources.